docs/validation-server/DESIGN.md

```diff
@@ -0,0 +1,27 @@
+The Plutus Core validation server will behave as follows:
+
+  - Bind to port 3000
+
+  - Accept HTTP requests with a JSON body
+
+  - The server should accept `GET` requests that contain the JSON data for
+    validation in the request body
+
+    - As an example:
+      ```json
+      {
+          "validator": "ab34ce2d",
+          "redeemer": "5cb418",
+          "data": "958bce"
+      }
+
+    - For other methods (e.g. `POST`), it will error (response status 400)
+
+  - The server will ignore HTTP headers
+
+  - The server will return response status 200 with no headers. The response
+    body will contain JSON: either `True` or `False`
+
+    ```json
+    { "isValid": true }
```

Comment:
We might as well return a richer error explaining what went wrong: we have at least two answers (didn't typecheck, evaluated with an error).

Reply:
We also need to return gas cost eventually